core/src/main/java/org/elasticsearch/index/shard/ShadowIndexShard.java

```diff
@@ -102,4 +103,22 @@ public boolean shouldFlush() {
     public TranslogStats translogStats() {
         return null; // shadow engine has no translog
     }
+
+
+    @Override
+    public void updateGlobalCheckpointOnReplica(long checkpoint) {
+        // nocommit: think shadow replicas through
```

Comment:
I assume the plan is to address this in a later PR (either way is fine, just want to know what the plan is)?